### PR TITLE
fix the parsing of addr

### LINKED ISSUE #1260: Errors in tracee-rules go signatures
When running tracee-ebpf with JSON output and feeding it to tracee-rules the default GO signatures fail to parse the addr argument:
```
2021/12/15 16:50:15 error handling event by signature Standard Input/Output Over Socket: couldn't convert arg to addr
```

### PR DESCRIPTION
**Pull request by AsafEitani**
fix #1260

### CODE REVIEW

**Review by danielpacak — CHANGES_REQUESTED** [review]

File: signatures/golang/stdio_over_socket.go [hunk 1]
```diff
@@ -213,7 +213,18 @@ func getAddressfromAddrArg(arg external.Argument) (connectedAddress, error) {
 
 	addr, isOk := arg.Value.(map[string]string)
 	if !isOk {
-		return connectedAddress{}, fmt.Errorf("couldn't convert arg to addr")
+		addr = make(map[string]string)
+		stringInterMap, isStringInterMap := arg.Value.(map[string]interface{})
```
[the remark this answers is not in the document]
Comment (roikol) [reply signatures/golang/stdio_over_socket.go:217]: hey,
this happens when using output option format:json. 
with format:gob the Value is `map[string]string` as expected.

[the remark this answers is not in the document]
Comment (yanivagman) [reply signatures/golang/stdio_over_socket.go:217]: If that's the case, the bug is that we have different outputs for json and gob formats, isn't it?

[the remark this answers is not in the document]
Comment (roikol) [reply signatures/golang/stdio_over_socket.go:217]: yes it is

[the remark this answers is not in the document]
Comment (yanivagman) [reply signatures/golang/stdio_over_socket.go:217]: So I think this is not the correct place to have the fix.
We should find the cause for the difference and handle it instead

[the remark this answers is not in the document]
Comment (yanivagman) [reply signatures/golang/stdio_over_socket.go:217]: @itaysk can it be related to #988 or something similar?

[the remark this answers is not in the document]
Comment (danielpacak) [reply signatures/golang/stdio_over_socket.go:217]: I'm not sure what is the goal of #988 but I had a look into code and basically GOB encodes and decodes `map[string]string` without loosing type information of map values. However, JSON unmarshaller decodes to `map[string]interface{}` and loose type information.

We can maybe patch it in existing, custom `UnmarshalJSON` callback, but I'm wondering if that's the only corner case to handle to keep the signature implementation agnostic of data serialisation format. WDYT?

```go
// UnmarshalJSON implements the json.Unmarshaler interface.
func (arg *Argument) UnmarshalJSON(b []byte) error {
	type argument Argument //alias Argument so we can unmarshal it within the unmarshaler implementation
	d := json.NewDecoder(bytes.NewReader(b))
	d.UseNumber()
	if err := d.Decode((*argument)(arg)); err != nil {
		return err
	}
	if arg.Value == nil {
		return nil
	}
	// Workaround to keep signature handler agnostic of GOB or JSON serialisation format
	if v, ok := arg.Value.(map[string]interface{}); ok && len(v) > 0 {
		mapWithStringValues := make(map[string]string, len(v))
		for key, value := range v {
			if stringValue, ok := value.(string); ok {
				mapWithStringValues[key] = stringValue
			}
		}
		arg.Value = mapWithStringValues
	}
```

**Comment by yanivagman** [comment]
most likely related to #988

**Comment by simar7** [comment]
Data point: Just loaded a fresh Tracee and I think the issue is still there.
```
tracee@e568d7d2e632[/tracee]$ sudo ./dist/tracee-ebpf \
  -o format:json \
  -o option:parse-arguments \
  --trace comm=bash \
  --trace follow \
  --trace event!='sched*' | \
  ./dist/tracee-rules \
  --input-tracee file:stdin \
  --input-tracee format:json
Loaded 14 signature(s): [TRC-1 TRC-13 TRC-2 TRC-14 TRC-3 TRC-11 TRC-9 TRC-4 TRC-5 TRC-12 TRC-8 TRC-6 TRC-10 TRC-7]
2022/04/13 21:01:31 error handling event by signature Standard Input/Output Over Socket: couldn't convert arg to addr
2022/04/13 21:01:31 error handling event by signature Standard Input/Output Over Socket: couldn't convert arg to addr
2022/04/13 21:01:31 error handling event by signature Standard Input/Output Over Socket: couldn't convert arg to addr
2022/04/13 21:01:31 error handling event by signature Standard Input/Output Over Socket: couldn't convert arg to addr
```

The trigger was me ssh-ing into the VM.

**Comment by rafaeldtinoco** [comment]
I'm moving this to draft and stale status until this is re-worked/re-discussed.

**Comment by yanivagman** [comment]
@AsafEitani I see that @NDStrahilevitz is making changes in SockAddr parsing logic. Do you still want to keep this PR opened or can we close it?

**Comment by AsafEitani** [comment]
Old PR

**Pull request closed without merging**